### PR TITLE
[NoQA] Chore: Analytics - split `SWITCH_REPORT` event

### PR DESCRIPTION
**Pull request by kacper-mikolajczak**
### Details

In this PR, `SWITCH_REPORT` is being divided into fine grained events in order to get better analytics result and fix event flooding issue.

Current changes are:

- [X] `SWITCH_REPORT_THREAD`
- [X] `SWITCH_REPORT_FROM_PREVIEW`
- [X] remove `SWITCH_REPORT` from `ReportActionItemParentAction` - it was probably the reason we've got so high timing results, because it was not properly closed
- [X] `REPORT_INITIAL_RENDER` removed - incorrect implementation

### Fixed Issues

$  https://github.com/Expensify/App/issues/46965
PROPOSAL: https://expensify.slack.com/archives/C05LX9D6E07/p1722451784265129

### Tests

**INTERNAL QA**

Check the App Timing dashboard and confirm these new events are reporting properly.

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by kacper-mikolajczak** [comment]
> SWITCH_REPORT_THREAD_GO_BACK - does not work because the previous screen is not re-rendered

Right now we only measured opening of the reports (different ones like threads, previews but still just openings). With THREAD_GO_BACK it's quite different because we are not unmounting previous screen, so onLayout is not triggered (on this event we are finishing the measurements).

:warning: Moreover, we are also leaking the event timing which is going to provide incorrect timings (might be the reason of reported high timings):

https://github.com/user-attachments/assets/08aa4a92-67ef-456c-93ab-8b2e90290ea1

With that in mind, I'd consider removing this event for now and consider improving current approach in general (e.g. event invalidation when we are out of some scope like a specific report)

CC @mountiny @adhorodyski

**Review by MrRefactor — APPROVED** [review]
Please fix linter, lgtm

**Comment by mountiny** [comment]
@hannojg do you use these flags in the E2E tests? Is it ok to change it from that perspective

**Comment by mountiny** [comment]
@kacper-mikolajczak if this is noqa can you please mark it as such in the title?

**Comment by hannojg** [comment]
@mountiny yeah I think that will break e2e tests - @kirillzyusko what do you think?

**Comment by kacper-mikolajczak** [comment]
While working on the PR I have found one occurrence of `TIMING.SWITCH_REPORT` in e2e test in `linkingTest.e2e.ts`:

https://github.com/Expensify/App/blob/0926267c22e50e68b52c055dacc3af775235ccd4/src/libs/E2E/tests/linkingTest.e2e.ts#L56-L82

The original `TIMING.SWITCH_REPORT` remains untouched so this flow should be fine.

If you could indeed check that @kirillzyusko. Much appreciated! ❤️

**Comment by rojiphil** [comment]
One curious observation is that we end the timer for `REPORT_INITIAL_RENDER` when we have not started the timer for the same. Though not directly related to the changes made thus far, did we miss adding the start timer for `REPORT_INITIAL_RENDER`?

https://github.com/Expensify/App/blob/0926267c22e50e68b52c055dacc3af775235ccd4/src/pages/home/report/ReportActionsView.tsx#L451

**Comment by adhorodyski** [comment]
> One curious observation is that we end the timer for `REPORT_INITIAL_RENDER` when we have not started the timer for the same. Though not directly related to the changes made thus far, did we miss adding the start timer for `REPORT_INITIAL_RENDER`?
> 
> https://github.com/Expensify/App/blob/0926267c22e50e68b52c055dacc3af775235ccd4/src/pages/home/report/ReportActionsView.tsx#L451

Wild take, but what about removing this one all together? I don't believe it's something with a super clear definition, to me it's more challenging to understand it and it's value than all the other ones that actually point to specific actions (we know when they start & finish).

**Comment by adhorodyski** [comment]
I think I'd also vote for getting rid of the `SWITCH_REPORT_THREAD_GO_BACK` - there's eg a native Android's back action that we don't include, and overall the 'Back' action will almost all the time just be fast because it's already on the stack (we're not dealing with the mount). What potentially lags here is just the animation which isn't something we care about (this much) in here.

Overall, I think we should track aggressively for the 'visiting' action in all scenarios, and skip the 'Back' (visit) which already happened with all the heavy operations that occurred the 1st time when mounting a Report screen.

**Comment by adhorodyski** [comment]
@mountiny curious of your take on this one

**Comment by mountiny** [comment]
@adhorodyski I like that, I think it makes sense to prioritize the visiting actions over going back.

> Timing.end(CONST.TIMING.REPORT_INITIAL_RENDER); 

Yeah, if we do not start it anywhere we can remove it, nobody seems to be using it so nobody will miss it.

**Comment by kacper-mikolajczak** [comment]
> [Wild take, but what about removing this one all together?](https://github.com/Expensify/App/pull/46966#issuecomment-2275803065)

That's also an option! @mountiny is there anyone who we could ask to address Adam's idea?

@rojiphil what do. you think of recent additions to the discussion?

File: src/pages/home/report/ReportActionItemParentAction.tsx [hunk 1]
```diff
@@ -135,7 +133,6 @@ function ReportActionItemParentAction({
                                               // Pop the chat report screen before navigating to the linked report action.
                                               Navigation.goBack(ROUTES.REPORT_WITH_ID.getRoute(ancestor.report.parentReportID ?? '-1', ancestor.reportAction.reportActionID));
                                           }
-                                          Timing.start(CONST.TIMING.SWITCH_REPORT);
```
Comment (adhorodyski) [inline src/pages/home/report/ReportActionItemParentAction.tsx:138]: Why do we have to remove this invocation?

File: src/pages/home/report/ThreadDivider.tsx [hunk 2]
```diff
@@ -48,7 +47,6 @@ function ThreadDivider({ancestor, isLinkDisabled = false}: ThreadDividerProps) {
             ) : (
                 <PressableWithoutFeedback
                     onPress={() => {
-                        Timing.start(CONST.TIMING.SWITCH_REPORT);
```
Comment (adhorodyski) [inline src/pages/home/report/ThreadDivider.tsx:51]: Same as before, isn't this the right place?

Comment (kacper-mikolajczak) [reply src/pages/home/report/ThreadDivider.tsx:51]: `ThreadDivider` is performs `goBack` action and it won't measure the report switching time correctly (the timer will remain active till another report is open). Here is a reply with more details https://github.com/Expensify/App/pull/46966#issuecomment-2273539824

**Comment by rojiphil** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

NA

</details>

<details>
<summary>MacOS: Desktop</summary>

NA

</details>

<details>
<summary>iOS: mWeb Safari</summary>

NA

</details>

<details>
<summary>Android: Native</summary>

NA

</details>

<details>
<summary>Android: mWeb Chrome</summary>

NA

</details>

<details>
<summary>iOS: Native</summary>

NA

</details>

File: src/pages/home/report/ReportActionItemParentAction.tsx (hunk 1 above)
Comment (kacper-mikolajczak) [reply src/pages/home/report/ReportActionItemParentAction.tsx:138]: Same case as https://github.com/Expensify/App/pull/46966/files#r1711083558

**Review by rojiphil — APPROVED** [review]
Code changes are straightforward. LGTM. Thanks.

File: src/pages/home/report/ThreadDivider.tsx (hunk 2 above)
Comment (adhorodyski) [reply src/pages/home/report/ThreadDivider.tsx:51]: oh clear, my bad! Thank you :)

**Comment by kacper-mikolajczak** [comment]
Hey folks, if you could take a look at this idea as well: https://github.com/Expensify/App/issues/47142 🙏

**Comment by rojiphil** [comment]
> Hey folks, if you could take a look at this idea as well: #47142 🙏

Well! Currently, it looks like the term `OPEN_REPORT` is used distinctly to refer to opening the report by clicking on the LHN row. At all other places, we are currently considering this as `SWITCH_REPORT`. If this distinction is meaningless, I think it makes sense to go ahead with a common usage of term `OPEN_REPORT`

https://github.com/Expensify/App/blob/55ea2f80b243874a3ebdedd67339d37f515a6735/src/components/LHNOptionsList/OptionRowLHN.tsx#L148

File: src/pages/home/report/ReportActionItemThread.tsx [hunk 3]
```diff
@@ -46,7 +46,7 @@ function ReportActionItemThread({numberOfReplies, icons, mostRecentReply, childR
             <PressableWithSecondaryInteraction
                 onPress={() => {
                     Report.navigateToAndOpenChildReport(childReportID);
-                    Timing.start(CONST.TIMING.SWITCH_REPORT);
+                    Timing.start(CONST.TIMING.SWITCH_REPORT_THREAD);
```
Comment (rojiphil) [inline src/pages/home/report/ReportActionItemThread.tsx:49]: And, by the way, why do we start the timer after initiating a navigation? Shouldn't we call `Timing.start` before `navigateToAndOpenChildReport`? And could this be the reason for the high timings in the graph?

Comment (mountiny) [reply src/pages/home/report/ReportActionItemThread.tsx:49]: I think we can discuss but its NAB for this PR

**Review by mountiny — APPROVED** [review]
Thanks!

File: src/pages/home/report/ReportActionItemThread.tsx (hunk 3 above)
Comment (rojiphil) [reply src/pages/home/report/ReportActionItemThread.tsx:49]: @mountiny Do you think we can create another issue for this and discuss? Or if there is another related issue on analytics we can add this issue there and discuss.

**Comment by mountiny** [comment]
This is now being sent 
image

**Pull request merged**